server/data/supabase_server_client.ts

```diff
@@ -0,0 +1,25 @@
+"use server";
+
+import { cookies } from "next/headers";
+import { createServerClient } from "@supabase/ssr";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { Database } from "@/data/supabaseTypes";
+
+export async function buildSupabaseServerClient(): Promise<
+  SupabaseClient<Database>
+> {
+  const cookieStore = cookies();
```

Comment:
maybe cookie store should be injected to allow for better testing

Reply:
ok, so I looked into this, but I'm not sure it will actually help with testing. the cookies are used to identify the user, as there are certain role level access limitations enforced at the database level.

we can't inject the cookies all the way from the page load, because this cookiestore is only available in a `"use server"` file (and it handles the clientside to serverside mapping of those cookies.)

if we move this one level up, it would be in the wrapper functions used to get specific items from the database (currently the `getQuery` function.) it would essentially result in having to include this line in any function that provides objects from the database, and still would require either an integration test with an actual request and cookies, or some sort of mocking. 

long story short, for now, I think this makes sense here. we can refactor later if we find a way to make it work better for testing.